File: src/test/java/io/icednut/algorithm/exercise/etc/leet884/SolutionTest.java
package io.icednut.algorithm.exercise.etc.leet884;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;

class SolutionTest {

    @Test
    void test01() {
        var solution = new Solution();

        String[] output = solution.uncommonFromSentences(
                "this apple is sweet",
                "this apple is sour"
        );

        assertArrayEquals(
                new String[]{"sweet", "sour"},
                output
        );
    }

    @Test
    void test02() {
        var solution = new Solution();

        String[] output = solution.uncommonFromSentences(
                "apple apple",
                "banana"
        );

        assertArrayEquals(
                new String[]{"banana"},
                output
        );
    }

    @Test
    void test03() {
        var solution = new Solution();

        String[] output = solution.uncommonFromSentences(
                "fo ly ly",
                "fo fo etx"
        );

        assertArrayEquals(
                new String[]{"etx"},
                output
        );
    }

}

File: src/test/kotlin/io/icednut/algorithm/exercise/greedy/leetcode1710/SolutionTest.kt
package io.icednut.algorithm.exercise.greedy.leetcode1710

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class SolutionTest {

    @Test
    fun test1() {
        val solution = Solution()

        val maximumUnits = solution.maximumUnits(arrayOf(intArrayOf(1, 3), intArrayOf(2, 2), intArrayOf(3, 1)), 4)

        assertEquals(8, maximumUnits)
    }

    @Test
    fun test2() {
        val solution = Solution()

        val maximumUnits = solution.maximumUnits(arrayOf(intArrayOf(5, 10), intArrayOf(2, 5), intArrayOf(4, 7), intArrayOf(3, 9)), 10)

        assertEquals(91, maximumUnits)
    }
}